lib/sneakers_handlers/retry_handler.rb

```diff
@@ -0,0 +1,93 @@
+# Using this handler, failed messages will be retried a certain number of times,
+# until they are dead-lettered.
+#
+# To use it you need to defined this handler in your worker:
+#
+# from_queue "my-app.queue_name",
+#   exchange: "domain_events",
+#   routing_key: "my_routing_key",
+#   handler: Sneakers::Handlers::RetryHandler,
+#   arguments: { "x-dead-letter-exchange" => "dlx.domain_events",
+#                "x-dead-letter-routing-key" => "my-app.queue_name" }}
+#
+# By default it will retry 5 times before dead-lettering a message, but you can
+# also customize that with the `max_retry` option:
+#
+# from_queue "my-app.queue_name",
+#   exchange: "domain_events",
+#   routing_key: "my_routing_key",
+#   handler: Sneakers::Handlers::RetryHandler,
+#   max_retry: 10,
+#   arguments: { "x-dead-letter-exchange" => "dlx.domain_events" }
+#                "x-dead-letter-routing-key" => "my-app.queue_name" }}
+
+module SneakersHandlers
+  class RetryHandler
+
+    def initialize(channel, queue, options)
+      @channel = channel
+      @queue = queue
+      @routing_key = options[:routing_key]
+      @max_retry = options.fetch(:max_retry, 5)
+
+      create_dlx(channel, queue, options)
+    end
+
+    def acknowledge(hdr, _props, _msg)
+      @channel.acknowledge(hdr.delivery_tag, false)
+    end
+
+    def reject(hdr, props, msg, requeue = false)
+      if requeue
```

Comment:
Should we remove this options? Doesn't seem to make sense to me if we are doing a custom thing.